kuma/javascript/src/header/search.jsx

```diff
@@ -65,8 +65,12 @@ export default function Search({ initialQuery }: Props) {
                 />
             </form>
             <button className="toggle-form" onClick={handleClick}>
-                <CloseIcon className="close-icon" />
-                <SearchIcon className="search-icon" />
+                <CloseIcon
+                    className={showForm ? 'close-icon' : 'close-icon hide'}
+                />
+                <SearchIcon
+                    className={showForm ? 'search-icon hidden' : 'search-icon'}
+                />
```

Comment:
Why are we doing this? Why have it in the DOM if the CSS sets `visibility: hidden;`?
There's probably an explanation. A11y maybe? or something. Either way, we need a code comment that explains in to the future reader why we're not using React of this. 

Secondly, if there's a good reason for using CSS instead of React, you probably have a typo in there. `hide  != hidden`.

Reply:
Also, have you considered doing `<SearchIcon style={!showForm && {visibility:'hidden'}}>` instead? 
Or, perhaps put this SCSS in a dedicate `search.scss` so that when/if we refactor this code we can't forget to undo the hack from such a fundamental file as `base/_utilities.scss` where it's likely to get stuck forever.

Reply:
> Why are we doing this? Why have it in the DOM if the CSS sets visibility: hidden;?

I cannot remember 100% why we kept it like this. I and Mindy did discuss this and there was a good reason for not swapping the actual icons. I will try and dig up the reasoning. It was not for accessibility reasons because we have the text 'Close search' and 'Open search'. I believe it has to do with the animation on interaction.

> Secondly, if there's a good reason for using CSS instead of React, you probably have a typo in there. hide != hidden.

This is intentional. The close icon is animated along with the form and so, we cannot use `display: none;` as transitions will not work. I, therefore, set the one to hidden(`display:none`) and the other to hide(`visibility:hidden`)

The utility classes `hidden` and `hide` will be/is being used elsewhere so, keeping them in utils makes sense.

Reply:
Yup, it was to enable animation:
https://github.com/mdn/kuma/pull/7052/files#r424314565